### ISSUE TITLE
7.18.2 regression: react/jsx-indent lots of false positives

### ISSUE THREAD

**Issue by martinpitt** [yannickcr/eslint-plugin-react#2563]
Since the last version 7.18.2, eslinting [cockpit](https://github.com/cockpit-project/cockpit/) gets tons of false positives like these:
```
/var/home/martin/upstream/cockpit/pkg/systemd/init.js
  1326:20  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1327:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1327:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1328:21  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
  1329:23  error  Expected indentation of 12 space characters but found 0  react/jsx-indent
```

on this code, which seems perfectly well indented:
```
    function create_timer_file() {
        var unit = "[Unit]\nDescription=";
        var service = "\n[Service]\nExecStart=";
        var timer = "\n[Timer]\n";
        var install = "[Install]\nWantedBy=timers.target\n";
        var service_file = unit + timer_unit.Description + service + timer_unit.Command + "\n";
```
The dozens of other errors are also wrong. Things were fine until 7.18.0, I haven't tested 7.18.1 yet.

**Comment by martinpitt** [comment]
More direct reproducer:
```sh
curl -O https://raw.githubusercontent.com/cockpit-project/cockpit/master/pkg/systemd/init.js
node_modules/.bin/eslint ./init.js
```

works fine with 7.18.0, fails with 7.18.2.

This file isn't that special, it affects a lot of other files in our tree. This smells like a rather generic parsing regression.

**Comment by ljharb** [comment]
cc @toshi-toma

**Comment by ljharb** [comment]
It should definitely only be matching jsxtext/literals inside jsx.

What version of eslint are you using?

**Comment by martinpitt** [comment]
We currently use 6.8.0, matching [package.json](https://github.com/cockpit-project/cockpit/blob/master/package.json)'s `"eslint": "^6.2.2"`.

That is the [current version](https://www.npmjs.com/package/eslint) AFAICS.

I copied the complete package-lock.json here: https://piware.de/tmp/package-lock.json

**Comment by toshi-toma** [comment]
@ljharb @martinpitt 

 I'm really sorry...
This bug is caused by my mistake.

Anyway, I created PR.(#2564 )
I'll under investigation for other cases.

**Comment by martinpitt** [comment]
@toshi-toma : Many thanks for the super-fast fix!
